backend/infrahub/events/node_action.py

```diff
@@ -37,14 +41,58 @@ def get_related(self) -> list[dict[str, str]]:
                 }
             )
 
-        for relationship_name in self.changelog.relationships.keys():
-            related.append(
-                {
-                    "prefect.resource.id": f"infrahub.node.{self.node_id}",
-                    "prefect.resource.role": "infrahub.node.field_update",
-                    "infrahub.field.name": relationship_name,
-                }
-            )
+        for relationship in self.changelog.relationships.values():
+            if isinstance(relationship, RelationshipCardinalityOneChangelog) and not relationship.is_empty:
+                if relationship.peer_id and relationship.peer_kind:
```

Comment:
not a blocker but it feels like there is some duplicated code here that could be consolidated

Reply:
I've cleaned this up a bit.